src/Compilers/Core/CodeAnalysisTest/DefaultAnalyzerAssemblyLoaderTests.cs

```diff
@@ -83,5 +118,194 @@ public void AssemblyLoading()
+            var g = gamma.CreateInstance("Gamma.G")!;
+            g.GetType().GetMethod("Write")!.Invoke(g, new object[] { sb, "Test G" });
+
+            Assembly epsilon = loader.LoadFromPath(_testFixture.Epsilon.Path);
+            var e = epsilon.CreateInstance("Epsilon.E")!;
+            var eWrite = e.GetType().GetMethod("Write")!;
+
+            var actual = sb.ToString();
+            if (ExecutionConditionUtil.IsCoreClr)
+            {
+                var exception = Assert.Throws<TargetInvocationException>(() => eWrite.Invoke(e, new object[] { sb, "Test E" }));
+                Assert.IsAssignableFrom<FileNotFoundException>(exception.InnerException);
+            }
+            else
+            {
+                eWrite.Invoke(e, new object[] { sb, "Test E" });
+                Assert.Equal(
+@"Delta: Gamma: Test G
+",
+                    actual);
+            }
+        }
+
+        [Fact]
+        public void AssemblyLoading_AnalyzerReferencesSystemCollectionsImmutable_01()
+        {
+            StringBuilder sb = new StringBuilder();
+
+            var loader = new DefaultAnalyzerAssemblyLoader();
+            loader.AddDependencyLocation(_testFixture.UserSystemCollectionsImmutable.Path);
+            loader.AddDependencyLocation(_testFixture.AnalyzerReferencesSystemCollectionsImmutable1.Path);
+
+            Assembly analyzerAssembly = loader.LoadFromPath(_testFixture.AnalyzerReferencesSystemCollectionsImmutable1.Path);
+            var analyzer = analyzerAssembly.CreateInstance("Analyzer")!;
+
+            if (ExecutionConditionUtil.IsCoreClr)
+            {
+                var ex = Assert.ThrowsAny<Exception>(() => analyzer.GetType().GetMethod("Method")!.Invoke(analyzer, new object[] { sb }));
+                Assert.True(ex is MissingMethodException or TargetInvocationException, $@"Unexpected exception type: ""{ex.GetType()}""");
+            }
+            else
+            {
+                analyzer.GetType().GetMethod("Method")!.Invoke(analyzer, new object[] { sb });
+                Assert.Equal("42", sb.ToString());
+            }
+        }
+
+        [Fact]
+        public void AssemblyLoading_AnalyzerReferencesSystemCollectionsImmutable_02()
+        {
+            StringBuilder sb = new StringBuilder();
+
+            var loader = new DefaultAnalyzerAssemblyLoader();
+            loader.AddDependencyLocation(_testFixture.UserSystemCollectionsImmutable.Path);
+            loader.AddDependencyLocation(_testFixture.AnalyzerReferencesSystemCollectionsImmutable2.Path);
+
+            Assembly analyzerAssembly = loader.LoadFromPath(_testFixture.AnalyzerReferencesSystemCollectionsImmutable2.Path);
+            var analyzer = analyzerAssembly.CreateInstance("Analyzer")!;
+            analyzer.GetType().GetMethod("Method")!.Invoke(analyzer, new object[] { sb });
+            Assert.Equal(ExecutionConditionUtil.IsCoreClr ? "1" : "42", sb.ToString());
+        }
+
```

Comment:
One test I think is missing is firing up two different `DefaultAnalyzerAssemblyLoader` and verifying they can load different assemblies that have the same simple name. That is effectively what the compiler server is going to be doing.

Reply:
Does this comment describe a different scenario than what is covered in `AssemblyLoading_MultipleVersions`?

Reply:
Yes. In that test you're creating one `DefaultAnalyzerAssemblyLoader` and loading two different versions of Delta into it and verifying that one fails. I'm saying create two `DefaultAnalyzerAssemblyLoader` instances and load `Delta1` into the first and `Delta2` into the second. That is effectively how long lived hosts on .NET Core will work (new loader per compilation) and want to verify that it's working and that the correct `Assembly` is loaded into each context

Reply:
Actually one other test you need is loading `Delta1` and `Delta2` into the same `DefaultAnalyzerAssemblyLoader` but from different directories. That should work just fine now. That is essentially our core analyzers with conflicting dependencies case.

Reply:
We have `AssemblyLoading_MultipleVersions` to test loading different versions of the same assembly into one loader. I added a test `AssemblyLoading_MultipleVersions_MultipleLoaders` to do the same but with each version in a different loader.